app/services/nomis/api_slot_correcter.rb

```diff
@@ -0,0 +1,35 @@
+module Nomis
+  class ApiSlotCorrecter
```

Comment:
This name feels a bit 'bondagey'. How about 'TimeGranuator' or 'TimeSnapper' --- just a thought :)